packages/location-management/src/components/LocationUnitView/index.tsx

```diff
@@ -112,85 +79,48 @@ export function parseTableData(hierarchy: ParsedHierarchyNode[]) {
   return data;
 }
 
-/** Gets the hierarchy of the location units
- *
- * @param {Array<LocationUnit>} location - array of location units to get hierarchy of
- * @param {string} accessToken - Access token to be used for requests
- * @param {string} opensrpBaseURL - base url
- * @returns {Promise<Array<RawOpenSRPHierarchy>>} array of RawOpenSRPHierarchy
- */
-export async function getHierarchy(
-  location: LocationUnit[],
-  accessToken: string,
-  opensrpBaseURL: string
-) {
-  const hierarchy: RawOpenSRPHierarchy[] = [];
-  for await (const loc of location) {
-    const serve = new OpenSRPService(accessToken, opensrpBaseURL, LOCATION_HIERARCHY);
-    const data = await serve.read(loc.id).then((response: RawOpenSRPHierarchy) => response);
-    hierarchy.push(data);
-  }
-
-  return hierarchy;
-}
-
 export const LocationUnitView: React.FC<Props> = (props: Props) => {
   const accessToken = useSelector((state) => getAccessToken(state) as string);
+
   const treeData = (useSelector((state) =>
     getAllHierarchiesArray(state)
   ) as unknown) as ParsedHierarchyNode[];
-  const dispatch = useDispatch();
   const [tableData, setTableData] = useState<TableData[]>([]);
   const [detail, setDetail] = useState<LocationDetailData | 'loading' | null>(null);
   const [currentParentChildren, setCurrentParentChildren] = useState<ParsedHierarchyNode[]>([]);
   const { opensrpBaseURL } = props;
 
-  useEffect(() => {
-    if (!treeData.length) {
-      getBaseTreeNode(accessToken, opensrpBaseURL)
-        .then((response) => {
-          dispatch(fetchLocationUnits(response));
-          getHierarchy(response, accessToken, opensrpBaseURL)
-            .then((hierarchy) => {
-              hierarchy.forEach((hier) => {
-                const processed = generateJurisdictionTree(hier);
-                dispatch(fetchAllHierarchies(processed.model));
-              });
-            })
-            .catch(() => sendErrorNotification('An error occurred'));
-        })
-        .catch(() => sendErrorNotification('An error occurred'));
-    }
-  }, [treeData, accessToken, dispatch, opensrpBaseURL]);
-
   useEffect(() => {
     const data = parseTableData(currentParentChildren.length ? currentParentChildren : treeData);
     setTableData(data);
   }, [treeData, currentParentChildren]);
 
-  if (!tableData.length || !treeData.length)
-    return (
-      <Spin
-        style={{
-          display: 'flex',
-          alignItems: 'center',
-          justifyContent: 'center',
-          minHeight: '85vh',
-        }}
-        size={'large'}
-      />
-    );
-
   return (
     <section className="layout-content">
       <Helmet>
         <title>{LOCATION_UNIT}</title>
       </Helmet>
+      {(!tableData.length || !treeData.length) && (
+        // This condition will become false one the tree data is loaded from the tree component
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        <>
+          <Spin
+            style={{
```

Comment:
The `Spin` is styled in the client, do we need custom style here?

Reply:
done